src/vendor/api/v1/views.py

```diff
@@ -46,6 +46,21 @@ def session_cart(self, request, offer):
             session_cart[offer_key]['quantity'] = 1
 
         return session_cart
+    
+    def add_offer_to_customer_profile_cart(self, customer_profile, offer):
+        cart = customer_profile.get_cart_or_checkout_cart()
+
+        if cart.status == InvoiceStatus.CHECKOUT:
+            cart.status = InvoiceStatus.CART
+            cart.save()
+
+        if customer_profile.has_product(offer.products.all()) and not offer.allow_multiple:
+            messages.info(self.request, _("You Have Already Purchased This Item"))
+        elif cart.order_items.filter(offer__products__in=offer.products.all()).count() and not offer.allow_multiple:
```

Comment:
not blockers: 
- optimization wise, i think that .exists() is preferable to .count() if we just want to check that the queryset is not empty
- if we save offer.products.all() to a variable, we can reuse it in the first two checks and save a query

Reply:
Awesome, I will apply those changes, thanks :)